bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/firmware/FirmwareImpl.java

```diff
@@ -254,18 +239,26 @@ private boolean hasRequiredModel(Thing thing) {
         }
     }
 
+    private boolean firmwareOnThingIsHighEnough(Thing thing) {
+        if (prerequisiteVersion == null) {
+            return true;
+        } else {
+            String firmwareOnThing = thing.getProperties().get(PROPERTY_FIRMWARE_VERSION);
+            return firmwareOnThing != null && new Version(firmwareOnThing).compare(prerequisiteVersion) >= 0;
+        }
+    }
+
     private static class Version {
 
         private static final int NO_INT = -1;
 
+        private final String versionString;
         private final String[] parts;
 
-        private Version(@Nullable String versionString) {
-            if (versionString == null) {
-                this.parts = new String[] {};
-            } else {
-                this.parts = versionString.split("-|_|\\.");
-            }
+        private Version(String versionString) {
+            ParameterChecks.checkNotNull(versionString, "versionString");
```

Comment:
Since this is a private class with a private constructor, I think it is sufficient to annotate the parameter via `@NonNullByDefault` and omit the check since it can only be called by framework code anyway, right? See https://www.eclipse.org/smarthome/documentation/development/conventions.html#null-annotations

Reply:
This sounds sensible, I will remove the `ParameterChecks...`.